File: 2002B2/CS/WinForms-Framework-POP3Checker/Pop3Checker/Pop3CheckerSettingsForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace Pop3Checker
{
	/// <summary>
	/// Summary description for Pop3CheckerSettingsForm.
	/// </summary>
	public class Pop3CheckerSettingsForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label HostNameLabel;
		private System.Windows.Forms.TextBox UserNameEntry;
		private System.Windows.Forms.Button OKAction;
		private System.Windows.Forms.NumericUpDown HostPortEntry;
		private System.Windows.Forms.NumericUpDown RefreshIntervalEntry;
		private System.Windows.Forms.GroupBox MiscellaneousGroup;
		private System.Windows.Forms.NumericUpDown ConnectionTimeoutEntry;
		private System.Windows.Forms.TextBox UserPasswordEntry;
		private System.Windows.Forms.Label UserPasswordLabel;
		private System.Windows.Forms.TextBox HostNameEntry;
		private System.Windows.Forms.Label RefreshIntervalLabel;
		private System.Windows.Forms.Label ConnectionTimeoutLabel;
		private System.Windows.Forms.GroupBox UserGroup;
		private System.Windows.Forms.Label UserNameLabel;
		private System.Windows.Forms.Label HostPortLabel;
		private System.Windows.Forms.Button ApplyAction;
		private System.Windows.Forms.Button CancelAction;
		private System.Windows.Forms.GroupBox HostGroup;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components;

		//Defaults
		private const int mcDefaultConnectionTimeout = 30;
		private const int mcDefaultRefreshInterval = 600;
		private const int mcDefaultHostPort = 110;
		private const string mcApplicationName = "POP3 Checker";

		public Pop3CheckerSettingsForm()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//Set up form's controls with default values.
			InitializeSettings();
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		public override void Dispose()
		{
			base.Dispose();
			components.Dispose();
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.HostNameLabel = new System.Windows.Forms.Label();
			this.UserNameEntry = new System.Windows.Forms.TextBox();
			this.OKAction = new System.Windows.Forms.Button();
			this.HostPortEntry = new System.Windows.Forms.NumericUpDown();
			this.RefreshIntervalEntry = new System.Windows.Forms.NumericUpDown();
			this.MiscellaneousGroup = new System.Windows.Forms.GroupBox();
			this.ConnectionTimeoutEntry = new System.Windows.Forms.NumericUpDown();
			this.UserPasswordEntry = new System.Windows.Forms.TextBox();
			this.UserPasswordLabel = new System.Windows.Forms.Label();
			this.HostNameEntry = new System.Windows.Forms.TextBox();
			this.RefreshIntervalLabel = new System.Windows.Forms.Label();
			this.ConnectionTimeoutLabel = new System.Windows.Forms.Label();
			this.UserGroup = new System.Windows.Forms.GroupBox();
			this.UserNameLabel = new System.Windows.Forms.Label();
			this.HostPortLabel = new System.Windows.Forms.Label();
			this.ApplyAction = new System.Windows.Forms.Button();
			this.CancelAction = new System.Windows.Forms.Button();
			this.HostGroup = new System.Windows.Forms.GroupBox();
			((System.ComponentModel.ISupportInitialize)(this.HostPortEntry)).BeginInit();
			((System.ComponentModel.ISupportInitialize)(this.RefreshIntervalEntry)).BeginInit();
			((System.ComponentModel.ISupportInitialize)(this.ConnectionTimeoutEntry)).BeginInit();
			this.HostNameLabel.AutoSize = true;
			this.HostNameLabel.Location = new System.Drawing.Point(8, 20);
			this.HostNameLabel.Size = new System.Drawing.Size(64, 13);
			this.HostNameLabel.TabIndex = 0;
			this.HostNameLabel.Text = "&Host Name:";
			this.UserNameEntry.Location = new System.Drawing.Point(92, 16);
			this.UserNameEntry.Size = new System.Drawing.Size(172, 20);
			this.UserNameEntry.TabIndex = 1;
			this.UserNameEntry.Text = "";
			this.OKAction.Location = new System.Drawing.Point(8, 248);
			this.OKAction.TabIndex = 3;
			this.OKAction.Text = "OK";
			this.OKAction.Click += new System.EventHandler(this.OKAction_Click);
			this.HostPortEntry.Location = new System.Drawing.Point(72, 44);
			this.HostPortEntry.Maximum = new System.Decimal(new int[] {65536,
																		  0,
																		  0,
																		  0});
			this.HostPortEntry.Size = new System.Drawing.Size(72, 20);
			this.HostPortEntry.TabIndex = 3;
			this.RefreshIntervalEntry.Increment = new System.Decimal(new int[] {300,
																				   0,
																				   0,
																				   0});
			this.RefreshIntervalEntry.Location = new System.Drawing.Point(164, 44);
			this.RefreshIntervalEntry.Maximum = new System.Decimal(new int[] {3600,
																				 0,
																				 0,
																				 0});
			this.RefreshIntervalEntry.Size = new System.Drawing.Size(72, 20);
			this.RefreshIntervalEntry.TabIndex = 3;
			this.RefreshIntervalEntry.Value = new System.Decimal(new int[] {1800,
																			   0,
																			   0,
																			   0});
			this.MiscellaneousGroup.Controls.AddRange(new System.Windows.Forms.Control[] {this.RefreshIntervalEntry,
																							 this.ConnectionTimeoutEntry,
																							 this.RefreshIntervalLabel,
																							 this.ConnectionTimeoutLabel});
			this.MiscellaneousGroup.Location = new System.Drawing.Point(8, 164);
			this.MiscellaneousGroup.Size = new System.Drawing.Size(276, 76);
			this.MiscellaneousGroup.TabIndex = 2;
			this.MiscellaneousGroup.TabStop = false;
			this.MiscellaneousGroup.Text = "Miscellaneous";
			this.ConnectionTimeoutEntry.Increment = new System.Decimal(new int[] {30,
																					 0,
																					 0,
																					 0});
			this.ConnectionTimeoutEntry.Location = new System.Drawing.Point(164, 16);
			this.ConnectionTimeoutEntry.Maximum = new System.Decimal(new int[] {300,
																				   0,
																				   0,
																				   0});
			this.ConnectionTimeoutEntry.Minimum = new System.Decimal(new int[] {30,
																				   0,
																				   0,
																				   0});
			this.ConnectionTimeoutEntry.Size = new System.Drawing.Size(72, 20);
			this.ConnectionTimeoutEntry.TabIndex = 1;
			this.ConnectionTimeoutEntry.Value = new System.Decimal(new int[] {30,
																				 0,
																				 0,
																				 0});
			this.UserPasswordEntry.Location = new System.Drawing.Point(92, 44);
			this.UserPasswordEntry.Size = new System.Drawing.Size(172, 20);
			this.UserPasswordEntry.TabIndex = 3;
			this.UserPasswordEntry.Text = "";
			this.UserPasswordLabel.AutoSize = true;
			this.UserPasswordLabel.Location = new System.Drawing.Point(8, 48);
			this.UserPasswordLabel.Size = new System.Drawing.Size(84, 13);
			this.UserPasswordLabel.TabIndex = 2;
			this.UserPasswordLabel.Text = "User Pass&word:";
			this.HostNameEntry.Location = new System.Drawing.Point(72, 16);
			this.HostNameEntry.Size = new System.Drawing.Size(192, 20);
			this.HostNameEntry.TabIndex = 1;
			this.HostNameEntry.Text = "";
			this.RefreshIntervalLabel.AutoSize = true;
			this.RefreshIntervalLabel.Location = new System.Drawing.Point(8, 48);
			this.RefreshIntervalLabel.Size = new System.Drawing.Size(138, 13);
			this.RefreshIntervalLabel.TabIndex = 2;
			this.RefreshIntervalLabel.Text = "Seconds &between Checks:";
			this.ConnectionTimeoutLabel.AutoSize = true;
			this.ConnectionTimeoutLabel.Location = new System.Drawing.Point(8, 20);
			this.ConnectionTimeoutLabel.Size = new System.Drawing.Size(160, 13);
			this.ConnectionTimeoutLabel.TabIndex = 0;
			this.ConnectionTimeoutLabel.Text = "&Seconds to Wait for Response:";
			this.UserGroup.Controls.AddRange(new System.Windows.Forms.Control[] {this.UserPasswordLabel,
																					this.UserPasswordEntry,
																					this.UserNameEntry,
																					this.UserNameLabel});
			this.UserGroup.Location = new System.Drawing.Point(8, 84);
			this.UserGroup.Size = new System.Drawing.Size(276, 76);
			this.UserGroup.TabIndex = 1;
			this.UserGroup.TabStop = false;
			this.UserGroup.Text = "User";
			this.UserNameLabel.AutoSize = true;
			this.UserNameLabel.Location = new System.Drawing.Point(8, 20);
			this.UserNameLabel.Size = new System.Drawing.Size(64, 13);
			this.UserNameLabel.TabIndex = 0;
			this.UserNameLabel.Text = "&User Name:";
			this.HostPortLabel.AutoSize = true;
			this.HostPortLabel.Location = new System.Drawing.Point(8, 48);
			this.HostPortLabel.Size = new System.Drawing.Size(54, 13);
			this.HostPortLabel.TabIndex = 2;
			this.HostPortLabel.Text = "Host &Port:";
			this.ApplyAction.Location = new System.Drawing.Point(208, 248);
			this.ApplyAction.TabIndex = 5;
			this.ApplyAction.Text = "&Apply";
			this.ApplyAction.Click += new System.EventHandler(this.ApplyAction_Click);
			this.CancelAction.DialogResult = System.Windows.Forms.DialogResult.Cancel;
			this.CancelAction.Location = new System.Drawing.Point(108, 248);
			this.CancelAction.TabIndex = 4;
			this.CancelAction.Text = "Cancel";
			this.CancelAction.Click += new System.EventHandler(this.CancelAction_Click);
			this.HostGroup.Controls.AddRange(new System.Windows.Forms.Control[] {this.HostPortEntry,
																					this.HostPortLabel,
																					this.HostNameEntry,
																					this.HostNameLabel});
			this.HostGroup.Location = new System.Drawing.Point(8, 4);
			this.HostGroup.Size = new System.Drawing.Size(276, 76);
			this.HostGroup.TabIndex = 0;
			this.HostGroup.TabStop = false;
			this.HostGroup.Text = "Host";
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(290, 279);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {this.CancelAction,
																		  this.ApplyAction,
																		  this.UserGroup,
																		  this.OKAction,
																		  this.HostGroup,
																		  this.MiscellaneousGroup});
			this.Text = "POP3 Checker Settings";
			((System.ComponentModel.ISupportInitialize)(this.HostPortEntry)).EndInit();
			((System.ComponentModel.ISupportInitialize)(this.RefreshIntervalEntry)).EndInit();
			((System.ComponentModel.ISupportInitialize)(this.ConnectionTimeoutEntry)).EndInit();

		}
		#endregion

		private void OKAction_Click(object sender, System.EventArgs e)
		{
			SaveSettings();
			this.Close();
		}

		private void CancelAction_Click(object sender, System.EventArgs e)
		{
			this.Close();
		}

		private void ApplyAction_Click(object sender, System.EventArgs e)
		{
			SaveSettings();
		}

		public string HostName
		{
			get
			{
				return this.HostNameEntry.Text;
			}
		}

		public string UserName
		{
			get
			{
				return this.UserNameEntry.Text;
			}
		}

		public string UserPassword
		{
			get
			{
				return this.UserPasswordEntry.Text;
			}
		}

		public int HostPort
		{
			get
			{
				return Convert.ToInt32(this.HostPortEntry.Value);
			}
		}

		public int ConnectionTimeout
		{
			get
			{
				return Convert.ToInt32(this.ConnectionTimeoutEntry.Value);
			}
		}

		public int RefreshInterval
		{
			get
			{
				return Convert.ToInt32(this.RefreshIntervalEntry.Value);
			}
		}

		//Initialize form's controls with default values.
		public void InitializeSettings()
		{
			this.HostPortEntry.Value = Convert.ToDecimal(mcDefaultHostPort);
			this.UserNameEntry.Text = SystemInformation.UserName;
			this.ConnectionTimeoutEntry.Value = Convert.ToDecimal(mcDefaultConnectionTimeout);
			this.RefreshIntervalEntry.Value = Convert.ToDecimal(mcDefaultRefreshInterval);
		}

		//Attempts to retrieve settings from the registry and copy them to the form's controls.
		//Returns success.
		public bool LoadSettings()
		{
			bool failed=false;
			Microsoft.Win32.RegistryKey key = Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Software", true).OpenSubKey(mcApplicationName, true);
			if (key == null)
			{
				failed = true;
			}
			else
			{
				object val;
				//Pattern: Attempt to get a registry value via its key and set
				//an object variable to the value. Check if the returned value
				//is nothing, which probably means the registry key is not present.
				//If it is nothing, indicate failure. Otherwise, set a form control
				//to the value.

				val = key.GetValue("Host Name");
		
				if (val == null)
				{
					failed = true;
				}
				else
				{
					this.HostNameEntry.Text = val.ToString();
				}
		
				val = key.GetValue("Host Port");

				if (val == null)
				{
					failed = true;
				}
				else	
				{
					this.HostPortEntry.Text = val.ToString();
				}
		
				val = key.GetValue("User Name");
				if (val ==null)
				{
					failed = true;
				}
				else
				{
					this.UserNameEntry.Text = val.ToString();
				}
				val = key.GetValue("User Password");
				if (val == null)
				{
					failed = true;
				}
				else
				{
					this.UserPasswordEntry.Text = val.ToString();
				}
		
				val = key.GetValue("Connection Timeout");
				if (val == null)
				{
					failed = true;
				}
				else
				{
					this.ConnectionTimeoutEntry.Text = val.ToString();
				}
				val = key.GetValue("Refresh Interval");
				if (val == null)
				{
					failed = true;
				}
				else
				{
					this.RefreshIntervalEntry.Text = val.ToString();
				}

				return !failed;
			}
			return false;
		}

		//Saves settings from the form's controls to the registry.
		public void SaveSettings()
		{
			//Cache the parent key because we'll need it twice if our key is not present.
			Microsoft.Win32.RegistryKey parentKey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Software", true);
			Microsoft.Win32.RegistryKey key = parentKey.OpenSubKey(mcApplicationName, true);

			if (key == null)
			{
				key = parentKey.CreateSubKey(mcApplicationName);
			}
			key.SetValue("Host Name", this.HostNameEntry.Text);
			key.SetValue("Host Port", this.HostPortEntry.Text);
			key.SetValue("User Name", this.UserNameEntry.Text);
			key.SetValue("User Password", this.UserPasswordEntry.Text);
			key.SetValue("Connection Timeout", this.ConnectionTimeoutEntry.Text);
			key.SetValue("Refresh Interval", this.RefreshIntervalEntry.Text);

		}
	}
}

File: 2002B2/VB/WinForms-Framework-POP3Checker/Pop3Checker/Pop3CheckerSettingsForm.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

'Dialog that retrieves settings from the registry, allows the user
'to change the settings, and saves the new settings back to the registry.
'This dialog is intended to be called modally.
Public Class Pop3CheckerSettingsForm
    Inherits System.Windows.Forms.Form
    
    'Defaults
    Private Const mcDefaultConnectionTimeout As Integer = 30
    Private Const mcDefaultRefreshInterval As Integer = 600
    Private Const mcDefaultHostPort As Integer = 110
    Private Const mcApplicationName As String = "POP3 Checker"
    
    Public Sub New()
        MyBase.New()
        
        'This call is required by the Win Form Designer.
        InitializeComponent()
        
        'Set up form's controls with default values.
        InitializeSettings()
    End Sub
    
    'Initialize form's controls with default values.
    Public Sub InitializeSettings()
        Me.HostPortEntry.Value = Convert.ToDecimal(mcDefaultHostPort)
        Me.UserNameEntry.Text = SystemInformation.UserName
        Me.ConnectionTimeoutEntry.Value = Convert.ToDecimal(mcDefaultConnectionTimeout)
        Me.RefreshIntervalEntry.Value = Convert.ToDecimal(mcDefaultRefreshInterval)
    End Sub
    
    'Attempts to retrieve settings from the registry and copy them to the form's controls.
    'Returns success.
    Public Function LoadSettings() As Boolean
        Dim failed As Boolean
        Dim key As Microsoft.Win32.RegistryKey = _
            Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Software", True).OpenSubKey(mcApplicationName, True)
        If key Is Nothing Then
            failed = True
        Else
            Dim val As Object
            With key
                'Pattern: Attempt to get a registry value via its key and set
                'an object variable to the value. Check if the returned value
                'is nothing, which probably means the registry key is not present.
                'If it is nothing, indicate failure. Otherwise, set a form control
                'to the value.
                
                val = .GetValue("Host Name")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.HostNameEntry.Text = val.ToString
                End If
                val = .GetValue("Host Port")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.HostPortEntry.Text = val.ToString
                End If
                val = .GetValue("User Name")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.UserNameEntry.Text = val.ToString
                End If
                val = .GetValue("User Password")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.UserPasswordEntry.Text = val.ToString
                End If
                val = .GetValue("Connection Timeout")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.ConnectionTimeoutEntry.Text = val.ToString
                End If
                val = .GetValue("Refresh Interval")
                If val Is Nothing Then
                    failed = True
                Else
                    Me.RefreshIntervalEntry.Text = val.ToString
                End If
            End With
            Return Not failed
        End If
    End Function
    
    'Saves settings from the form's controls to the registry.
    Public Sub SaveSettings()
        'Cache the parent key because we'll need it twice if our key is not present.
        Dim parentKey As Microsoft.Win32.RegistryKey = _
            Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Software", True)
        Dim key As Microsoft.Win32.RegistryKey = parentKey.OpenSubKey(mcApplicationName, True)
        
        If key Is Nothing Then
            key = parentKey.CreateSubKey(mcApplicationName)
        End If
        
        With key
            .SetValue("Host Name", Me.HostNameEntry.Text)
            .SetValue("Host Port", Me.HostPortEntry.Text)
            .SetValue("User Name", Me.UserNameEntry.Text)
            .SetValue("User Password", Me.UserPasswordEntry.Text)
            .SetValue("Connection Timeout", Me.ConnectionTimeoutEntry.Text)
            .SetValue("Refresh Interval", Me.RefreshIntervalEntry.Text)
        End With
    End Sub
    
    'Executes when the Cancel button is pressed.
    'Closes the form.
    Protected Sub CancelAction_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles CancelAction.click
        Me.Close()
    End Sub

    'Executes when the Apply button is pressed.
    'Saves the settings to the registry.
    Protected Sub ApplyAction_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles ApplyAction.click
        SaveSettings()
    End Sub

    'Executes when the OK button is pressed.
    'Saves the settings to the registry and closes the form.
    Protected Sub OKAction_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles OKAction.click
        SaveSettings()
        Me.Close()
    End Sub

#Region " Read-only properties "
    'Pattern: Read-only properties return form control values.

    ReadOnly Property HostName() As String
        Get
            Return Me.HostNameEntry.Text
        End Get
    End Property

    ReadOnly Property HostPort() As Integer
        Get
            Return Convert.ToInt32(Me.HostPortEntry.Value)
        End Get
    End Property

    ReadOnly Property UserName() As String
        Get
            Return Me.UserNameEntry.Text
        End Get
    End Property

    ReadOnly Property UserPassword() As String
        Get
            Return Me.UserPasswordEntry.Text
        End Get
    End Property

    ReadOnly Property ConnectionTimeout() As Integer
        Get
            Return Convert.ToInt32(Me.ConnectionTimeoutEntry.Value)

        End Get
    End Property

    ReadOnly Property RefreshInterval() As Integer
        Get
            Return Convert.ToInt32(Me.RefreshIntervalEntry.Value)
        End Get
    End Property

    'Form overrides dispose to clean up the component list.
    Public Overloads Overrides Sub Dispose()
        MyBase.Dispose()
        components.Dispose()
    End Sub
#End Region

#Region " Windows Form Designer generated code "

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.Container



    Private WithEvents HostGroup As System.Windows.Forms.GroupBox
    Private WithEvents UserGroup As System.Windows.Forms.GroupBox
    Private WithEvents MiscellaneousGroup As System.Windows.Forms.GroupBox

    Private WithEvents HostNameLabel As System.Windows.Forms.Label
    Private WithEvents HostPortLabel As System.Windows.Forms.Label
    Private WithEvents UserNameLabel As System.Windows.Forms.Label
    Private WithEvents UserPasswordLabel As System.Windows.Forms.Label
    Private WithEvents ConnectionTimeoutLabel As System.Windows.Forms.Label
    Private WithEvents RefreshIntervalLabel As System.Windows.Forms.Label

    Private WithEvents HostNameEntry As System.Windows.Forms.TextBox
    Private WithEvents HostPortEntry As System.Windows.Forms.NumericUpDown
    Private WithEvents UserNameEntry As System.Windows.Forms.TextBox
    Private WithEvents UserPasswordEntry As System.Windows.Forms.TextBox
    Private WithEvents ConnectionTimeoutEntry As System.Windows.Forms.NumericUpDown
    Private WithEvents RefreshIntervalEntry As System.Windows.Forms.NumericUpDown

    Private WithEvents OKAction As System.Windows.Forms.Button
    Private WithEvents CancelAction As System.Windows.Forms.Button
    Private WithEvents ApplyAction As System.Windows.Forms.Button

    Dim WithEvents Pop3Checker As System.Windows.Forms.Form

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Private Sub InitializeComponent()
        Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(Pop3CheckerSettingsForm))

        Me.components = New System.ComponentModel.Container()
        Me.ConnectionTimeoutLabel = New System.Windows.Forms.Label()
        Me.UserGroup = New System.Windows.Forms.GroupBox()
        Me.UserPasswordLabel = New System.Windows.Forms.Label()
        Me.HostNameEntry = New System.Windows.Forms.TextBox()
        Me.UserNameLabel = New System.Windows.Forms.Label()
        Me.CancelAction = New System.Windows.Forms.Button()
        Me.HostGroup = New System.Windows.Forms.GroupBox()
        Me.RefreshIntervalLabel = New System.Windows.Forms.Label()
        Me.ApplyAction = New System.Windows.Forms.Button()
        Me.UserPasswordEntry = New System.Windows.Forms.TextBox()
        Me.UserNameEntry = New System.Windows.Forms.TextBox()
        Me.OKAction = New System.Windows.Forms.Button()
        Me.HostNameLabel = New System.Windows.Forms.Label()
        Me.HostPortEntry = New System.Windows.Forms.NumericUpDown()
        Me.MiscellaneousGroup = New System.Windows.Forms.GroupBox()
        Me.ConnectionTimeoutEntry = New System.Windows.Forms.NumericUpDown()
        Me.RefreshIntervalEntry = New System.Windows.Forms.NumericUpDown()
        Me.HostPortLabel = New System.Windows.Forms.Label()

        HostPortEntry.BeginInit()
        ConnectionTimeoutEntry.BeginInit()
        RefreshIntervalEntry.BeginInit()

        '@design Me.TrayHeight = 90
        '@design Me.TrayLargeIcon = False
        '@design Me.TrayAutoArrange = True
        '@design Me.GridSize = New System.Drawing.Size(4, 4)
        ConnectionTimeoutLabel.Location = New System.Drawing.Point(8, 20)
        ConnectionTimeoutLabel.Text = "&Seconds to Wait for Response:"
        ConnectionTimeoutLabel.Size = New System.Drawing.Size(155, 13)
        ConnectionTimeoutLabel.AutoSize = True
        ConnectionTimeoutLabel.TabIndex = 0

        UserGroup.Location = New System.Drawing.Point(8, 84)
        UserGroup.TabIndex = 1
        UserGroup.TabStop = False
        UserGroup.Text = "User"
        UserGroup.Size = New System.Drawing.Size(276, 76)

        UserPasswordLabel.Location = New System.Drawing.Point(8, 48)
        UserPasswordLabel.Text = "User Pass&word:"
        UserPasswordLabel.Size = New System.Drawing.Size(81, 13)
        UserPasswordLabel.AutoSize = True
        UserPasswordLabel.TabIndex = 2

        HostNameEntry.Location = New System.Drawing.Point(72, 16)
        HostNameEntry.TabIndex = 1
        HostNameEntry.Size = New System.Drawing.Size(192, 20)

        UserNameLabel.Location = New System.Drawing.Point(8, 20)
        UserNameLabel.Text = "&User Name:"
        UserNameLabel.Size = New System.Drawing.Size(63, 13)
        UserNameLabel.AutoSize = True
        UserNameLabel.TabIndex = 0

        CancelAction.Location = New System.Drawing.Point(108, 248)
        CancelAction.DialogResult = System.Windows.Forms.DialogResult.Cancel
        CancelAction.Size = New System.Drawing.Size(75, 23)
        CancelAction.TabIndex = 4
        CancelAction.Text = "Cancel"

        HostGroup.Location = New System.Drawing.Point(8, 4)
        HostGroup.TabIndex = 0
        HostGroup.TabStop = False
        HostGroup.Text = "Host"
        HostGroup.Size = New System.Drawing.Size(276, 76)

        RefreshIntervalLabel.Location = New System.Drawing.Point(8, 48)
        RefreshIntervalLabel.Text = "Seconds &between Checks:"
        RefreshIntervalLabel.Size = New System.Drawing.Size(134, 13)
        RefreshIntervalLabel.AutoSize = True
        RefreshIntervalLabel.TabIndex = 2

        ApplyAction.Location = New System.Drawing.Point(208, 248)
        ApplyAction.Size = New System.Drawing.Size(75, 23)
        ApplyAction.TabIndex = 5
        ApplyAction.Text = "&Apply"

        UserPasswordEntry.Location = New System.Drawing.Point(92, 44)
        UserPasswordEntry.PasswordChar = ChrW(8226)
        UserPasswordEntry.TabIndex = 3
        UserPasswordEntry.Size = New System.Drawing.Size(172, 20)

        UserNameEntry.Location = New System.Drawing.Point(92, 16)
        UserNameEntry.TabIndex = 1
        UserNameEntry.Size = New System.Drawing.Size(172, 20)

        OKAction.Location = New System.Drawing.Point(8, 248)
        OKAction.Size = New System.Drawing.Size(75, 23)
        OKAction.TabIndex = 3
        OKAction.Text = "OK"

        HostNameLabel.Location = New System.Drawing.Point(8, 20)
        HostNameLabel.Text = "&Host Name:"
        HostNameLabel.Size = New System.Drawing.Size(62, 13)
        HostNameLabel.AutoSize = True
        HostNameLabel.TabIndex = 0

        HostPortEntry.Location = New System.Drawing.Point(72, 44)
        HostPortEntry.Maximum = New Decimal(65536#)
        HostPortEntry.Size = New System.Drawing.Size(72, 20)
        HostPortEntry.TabIndex = 3

        MiscellaneousGroup.Location = New System.Drawing.Point(8, 164)
        MiscellaneousGroup.TabIndex = 2
        MiscellaneousGroup.TabStop = False
        MiscellaneousGroup.Text = "Miscellaneous"
        MiscellaneousGroup.Size = New System.Drawing.Size(276, 76)

        ConnectionTimeoutEntry.Value = New Decimal(30#)
        ConnectionTimeoutEntry.Location = New System.Drawing.Point(164, 16)
        ConnectionTimeoutEntry.Maximum = New Decimal(300#)
        ConnectionTimeoutEntry.Size = New System.Drawing.Size(72, 20)
        ConnectionTimeoutEntry.Minimum = New Decimal(30#)
        ConnectionTimeoutEntry.TabIndex = 1
        ConnectionTimeoutEntry.Increment = New Decimal(30#)

        RefreshIntervalEntry.Value = New Decimal(1800#)
        RefreshIntervalEntry.Location = New System.Drawing.Point(164, 44)
        RefreshIntervalEntry.Maximum = New Decimal(3600#)
        RefreshIntervalEntry.Size = New System.Drawing.Size(72, 20)
        RefreshIntervalEntry.TabIndex = 3
        RefreshIntervalEntry.Increment = New Decimal(300#)

        HostPortLabel.Location = New System.Drawing.Point(8, 48)
        HostPortLabel.Text = "Host &Port:"
        HostPortLabel.Size = New System.Drawing.Size(53, 13)
        HostPortLabel.AutoSize = True
        HostPortLabel.TabIndex = 2
        Me.Text = "POP3 Checker Settings"
        Me.MaximizeBox = False
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.CancelButton = CancelAction
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Me.Icon = CType(resources.GetObject("$this.Icon"), System.Drawing.Icon)
        Me.AcceptButton = OKAction
        Me.CausesValidation = False
        Me.MinimizeBox = False
        Me.ClientSize = New System.Drawing.Size(290, 279)

        Me.Controls.Add(ApplyAction)
        Me.Controls.Add(CancelAction)
        Me.Controls.Add(OKAction)
        Me.Controls.Add(HostGroup)
        Me.Controls.Add(UserGroup)
        Me.Controls.Add(MiscellaneousGroup)
        MiscellaneousGroup.Controls.Add(RefreshIntervalEntry)
        MiscellaneousGroup.Controls.Add(ConnectionTimeoutEntry)
        MiscellaneousGroup.Controls.Add(RefreshIntervalLabel)
        MiscellaneousGroup.Controls.Add(ConnectionTimeoutLabel)
        UserGroup.Controls.Add(UserPasswordLabel)
        UserGroup.Controls.Add(UserPasswordEntry)
        UserGroup.Controls.Add(UserNameEntry)
        UserGroup.Controls.Add(UserNameLabel)
        HostGroup.Controls.Add(HostPortEntry)
        HostGroup.Controls.Add(HostPortLabel)
        HostGroup.Controls.Add(HostNameEntry)
        HostGroup.Controls.Add(HostNameLabel)

        HostPortEntry.EndInit()
        ConnectionTimeoutEntry.EndInit()
        RefreshIntervalEntry.EndInit()
    End Sub

#End Region

End Class
